### DISCUSSION TITLE
Work with MPAS

### DISCUSSION THREAD

**Discussion by mbranson** [NCAR/ADF#259] [Q&A]
Does ADF work with CESM history file that use the MPAS dycore?

**Comment by brianpm** [comment]
Hi Mark. Right now everything in ADF assumes the data is on a regular lat/lon grid. So for SE or MPAS experiments, we either write the output to a lat/lon grid or do a regridding step before running through ADF. 

Longer term we'd like to handle SE & MPAS better. There's already a *tiny* bit of code that knows how to handle and `ncol` dimension, but it's going to be a while before we can really handle irregular grids end-to-end.

**Reply by mbranson** [reply]
Thanks, Brian.  So should it work if I first remap the MPAS history file to the FV grid?  I tried that as well but then it crashed attempting to do the vertical interpolation to the constant pressure levels.

**Reply by brianpm** [reply]
Hmm. I thought we had dealt with this by checking for the variable `PMID` in the history files, and using that as the pressure field. Maybe @nusbaume can chime in.

**Reply by mbranson** [reply]
I thought the problem might be in adf_diag.py before it checks for PMID it already adds hyam,hybm,hyai,hybi to ncrcat_var_list and then later on obviously it can't find those so it crashes.  I tried hacking it with this:

                            if 'PMID' in hist_file_var_list:
                                ### mdb hack ###
                                ###ncrcat_var_list = ncrcat_var_list + ",PMID"
                                ncrcat_var_list = f"{var}"+",PS,PMID"
                                print("Adding PMID to file")

but I still get errors.

**Reply by nusbaume** [reply]
Yeah, it looks like before checking the vertical coordinate type it automatically tries to add the hybrid coordinate variables, as can be seen here:

https://github.com/NCAR/ADF/blob/main/lib/adf_diag.py#L548

I think if you either just comment-out that line, or put it inside a `vert_coord_type` check to skip it if you have a `height` vertical coordinate, then it should hopefully get you past your current error.  Of course if that doesn't work, or if you run into a new set of errors, then just let us know.  Thanks!
